### DISCUSSION TITLE
What happens to user data when changing from logging in user/password to OAuth?

### DISCUSSION THREAD

**Discussion by chrislawso** [immich-app/immich#10577] [Q&A]
I have never configured a selfhosted app with OAuth sign in and want to know how simple is the process of the same user having access to their pre-existing data in imich after they start to use OAuth.

If Immich is initially setup with logging in using username and password and then later users can also use OAuth then at what step in the process are the pre-existing user data mapped to the Oauth user? 

For example, if user Bob can previously used a username login to Immich and later logs in with OAuth with ie Authentik or Keycloak or whatever how does Immich know when Bob logs in with Authentik that it should show Bob's username data to Bob using Oauth?

Does the administrator have to perform any modifications to assign or associate linking Oauth users to pre-existing usernames? Does this need to be performed on the Oauth provider admin side or somewhere in immich admin?

Thank you

**Comment by bo0tzz** [comment]
The user gets linked based on the email, or can be linked manually from the user's settings.

**Comment by chrislawso** [comment]
After I setup the SSO and tested further I can confirm that users are automatically linked if they use the same email in both systems, idp and immich. 

However, if they do not have the same email unfortunately I cannot find how to manually link a user in user's settings.

**Comment by alexstrassheim** [comment]
If I try to use the same account, I receive the message "User already exists, but is linked to another account."

![linked-user](https://github.com/user-attachments/assets/011cdf38-d91b-41a5-8d50-f36263220c9a)

**Reply by alexstrassheim** [reply]
The fix for me was really simple. If you log in with email/password, you can manually unlink and link again it from **user** settings > oauth > OAuth link.
![oauth-link](https://github.com/user-attachments/assets/559f67fe-439e-4aff-b393-3516f498fb02)

**Reply by brnl** [reply]
> The fix for me was really simple. If you log in with email/password, you can manually unlink and link again it from **user** settings > oauth > OAuth link.
[image]

Should this also work from an administrative point? The reason I ask this I want to switch from Authentik to Keycloak. I recreate all users with the same name and e-mail. But asking them all to unlink (or create a password) before I can transfer is a bit much. Is there a central way to unlink all OAuth Connections?
